adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,411 @@
+            val splitRight = constraints.constrainWidth(splitArea.right.roundToInt())
+            val splitTop = constraints.constrainHeight(splitArea.top.roundToInt())
+            val splitBottom = constraints.constrainHeight(splitArea.bottom.roundToInt())
+            val firstConstraints =
+                if (isHorizontalSplit) {
+                    val width = when (layoutDirection) {
+                        LayoutDirection.Ltr -> splitLeft
+                        LayoutDirection.Rtl -> constraints.maxWidth - splitRight
+                    }
+
+                    constraints.copy(minWidth = width, maxWidth = width)
+                } else {
+                    constraints.copy(minHeight = splitTop, maxHeight = splitTop)
+                }
+            val secondConstraints =
+                if (isHorizontalSplit) {
+                    val width = when (layoutDirection) {
+                        LayoutDirection.Ltr -> constraints.maxWidth - splitRight
+                        LayoutDirection.Rtl -> splitLeft
+                    }
+                    constraints.copy(minWidth = width, maxWidth = width)
+                } else {
+                    val height = constraints.maxHeight - splitBottom
+                    constraints.copy(minHeight = height, maxHeight = height)
+                }
+            val firstPlaceable = firstMeasurable.measure(constraints.constrain(firstConstraints))
+            val secondPlaceable = secondMeasurable.measure(constraints.constrain(secondConstraints))
+
+            firstPlaceable.placeRelative(0, 0)
+            val detailOffsetX =
+                if (isHorizontalSplit) {
+                    constraints.maxWidth - secondPlaceable.width
+                } else {
+                    0
+                }
+            val detailOffsetY =
+                if (isHorizontalSplit) {
+                    0
+                } else {
+                    constraints.maxHeight - secondPlaceable.height
+                }
+            secondPlaceable.placeRelative(detailOffsetX, detailOffsetY)
+        }
+    }
+}
+
+/**
+ * A strategy for configuring the [TwoPane] component, that is responsible for the meta-data
+ * corresponding to the arrangement of the two panes of the layout.
+ */
+@Stable
+fun interface TwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds.
+     */
+    fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult
+}
+
+/**
+ * Returns the specification for where to place a split in [TwoPane] as a result of
+ * [TwoPaneStrategy.calculateSplitResult]
+ */
+class SplitResult(
+
+    /**
+     * Whether the split is vertical or horizontal
+     */
+    val isHorizontalSplit: Boolean,
+
+    /**
+     * The area that is nether a `start` pane or an `end` pane, but a separation between those two.
```

Comment:
Nice docs. We need to specify that this Rect is window-local, not layout local as well

Reply:
Done!